Support for sensors on cc3220sf launchxl rev c

anthonybrandon opened the pull request:
The cc3220sf launchpad revision C board has sensors for which zephyr already has drivers, unlike A and B revision. In this change I would like to support the rev. c board so it can work with the sensor_shell sample application. Since the A and B revisions have different sensors I didn't think just adding an overlay in the sample's boards dir would be a good solution.

One thing to note: according to the documentation for the rev. c board it still has the bma222e sensor, but this seems to be incorrect. If I read the chip id register it responds with the chip id for bma280. The two are almost entirely compatible, so I think this might be an oversight in the updated docs.

I added the new board target instead of just updating the existing one because I assume that one is intended to support the A or B revision.

anthonybrandon committed:
boards: cc3220sf_launchxl: update doc for rev.c

Document the support for temperature and accelerometer on the rev.c
board.

Signed-off-by: Anthony Brandon <anthony@amarulasolutions.com>

vanti reviewed (commented):
@anthonybrandon If you are going with this approach, you would have to add a config file for every sample that cc3220sf_launchxl supports - for example, samples/net/wifi no longer works when building for rev c. Otherwise you lose support for all of them when building for cc3220_launchxl_rev_c, which is a regression from the current state.

In my opinion adding an overlay in the sample's boards dir for the sensors required is not such a bad idea in comparison, as it is fairly low-touch and requires fewer files ie. less maintenance. Yes there is a risk someone with an old board might try to build the sample and the code won't run properly, but it is ok if we update the documentation as you do to state the difference between the board revisions. Once they read the docs they can then go get the latest revision if the sensor is needed. Also the probability of this happening decreases over time as the older boards disappear from the market, and it is nice to continue to compile using "west build -b cc3220sf_launchxl" with the latest board revision. 

It is also worth noting that the TI SimpleLink SDK follows the same approach and generally doesn't treat rev C as a different board from previous revisions.

anthonybrandon commented:
@vanti 

Thanks. I wasn't sure if that was acceptable. I will switch to using the overlay.

anthonybrandon committed:
samples: sensor_shell: add overlay for cc3220sf_launchxl

Enable the temperature and accelerometer sensors for the
cc3220sf_launchxl in the sensor_shell sample. These ones are only in the
C revision of the board. The A and B revisions have different sensors
not currently supported by zephyr.

Signed-off-by: Anthony Brandon <anthony@amarulasolutions.com>

vanti reviewed (approved):
Thanks for the change. The sample works for me.
